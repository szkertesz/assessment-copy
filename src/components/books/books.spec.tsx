import React from 'react';
import { render, screen } from '@testing-library/react';
import Books from './books';
import {  MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Store, AnyAction } from '@reduxjs/toolkit';

const mockStore = configureStore([]);
let store: Store<any, AnyAction>;

beforeEach(() => {
    store = mockStore({
        books: {
            books: [
                {
                    id: 1,
                    title: "Harry Potter and the Philosopher's Stone",
                    author: 'J. K. Rowlingswsdfsdf',
                    description:
                        "Harry Potter and the Philosopher's Stone is a fantasy novel written by British author J. K. Rowling. The first novel in the Harry Potter series and Rowling's debut novel, it follows Harry Potter, a young wizard who discovers his magical heritage on his eleventh birthday,",
                    reading: true,
                    read: false,
                },
            ],
            status: 'succeeded',
            error: null,
            filterOptions: {
                status: undefined,
            },
        },
    });
  });
describe('Books', () => {
    it('should render component', () => {
        const {container} = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Books />
                </MemoryRouter>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
    it('should render one component', () => {
        const {container} = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Books />
                </MemoryRouter>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
