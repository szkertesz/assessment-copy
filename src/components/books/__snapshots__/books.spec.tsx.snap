// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books should render component 1`] = `
<div>
  <header>
    <h1>
      Reading List
    </h1>
    <section
      class="action-bar"
    >
      <a
        href="/books/new"
      >
        New
      </a>
      <div />
      <fieldset
        class="status-toggle"
      >
        <legend>
          Filters:
        </legend>
        <div
          class="status-toggle__group"
        >
          <input
            checked=""
            class="visually-hidden"
            id="radio-reading"
            name="status-"
            type="radio"
            value="reading"
          />
          <label
            class="status-toggle__label button"
            for="radio-reading"
          >
            Reading
          </label>
        </div>
        <div
          class="status-toggle__group"
        >
          <input
            class="visually-hidden"
            id="radio-read"
            name="status-"
            type="radio"
            value="read"
          />
          <label
            class="status-toggle__label button"
            for="radio-read"
          >
            Read
          </label>
        </div>
      </fieldset>
    </section>
  </header>
  <main>
    <ul
      class="books__list"
    >
      <li>
        <article
          class="book-item"
        >
          <h2
            class="book-item__title"
          >
            Harry Potter and the Philosopher's Stone
          </h2>
          <div
            class="book-item__actions"
          >
            <a
              class="book-item__button button"
              href="/edit/1"
            >
              Edit
            </a>
            <fieldset
              class="status-toggle"
            >
              <div
                class="status-toggle__group"
              >
                <input
                  checked=""
                  class="visually-hidden"
                  id="1radio-reading"
                  name="status-1"
                  type="radio"
                  value="reading"
                />
                <label
                  class="status-toggle__label button"
                  for="1radio-reading"
                >
                  Reading
                </label>
              </div>
              <div
                class="status-toggle__group"
              >
                <input
                  class="visually-hidden"
                  id="1radio-read"
                  name="status-1"
                  type="radio"
                  value="read"
                />
                <label
                  class="status-toggle__label button"
                  for="1radio-read"
                >
                  Read
                </label>
              </div>
            </fieldset>
          </div>
        </article>
      </li>
    </ul>
  </main>
</div>
`;
